File: .curtain/go/1-cluster/main.go
package main

import (
	"github.com/pulumi/pulumi-civo/sdk/v2/go/civo"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		firewall, err := civo.NewFirewall(ctx, "firewall", &civo.FirewallArgs{
			Region:             pulumi.String("NYC1"),
			CreateDefaultRules: pulumi.Bool(true),
		})
		if err != nil {
			return err
		}
		cluster, err := civo.NewKubernetesCluster(ctx, "cluster", &civo.KubernetesClusterArgs{
			Region:     pulumi.String("NYC1"),
			FirewallId: firewall.ID(),
			Pools: &KubernetesClusterPoolsArgs{
				NodeCount: pulumi.Int(3),
				Size:      pulumi.String("g4s.kube.medium"),
			},
		})
		if err != nil {
			return err
		}
		ctx.Export("clusterName", cluster.Name)
		return nil
	})
}

File: .curtain/python/1-cluster/__main__.py
import pulumi
import pulumi_civo as civo

firewall = civo.Firewall("firewall",
    region="NYC1",
    create_default_rules=True)
cluster = civo.KubernetesCluster("cluster",
    region="NYC1",
    firewall_id=firewall.id,
    pools=civo.KubernetesClusterPoolsArgs(
        node_count=3,
        size="g4s.kube.medium",
    ))
pulumi.export("clusterName", cluster.name)
